src/v/net/server.cc

```diff
@@ -264,45 +267,46 @@ server::accept_finish(ss::sstring name, ss::future<ss::accept_result> f_cs_sa) {
       this->name(),
       ar.remote_address,
       name);
-    if (_conn_gate.is_closed()) {
-        co_await conn->shutdown();
```

Comment:
We don't need to do a shutdown here anymore. Since `shutdown_input` waits on _accept_gate it will correctly shutdown _all_ connections that ever arrived.